src/adapter/debugAdapter.ts

```diff
@@ -262,6 +264,48 @@ export class DebugAdapter {
     return {};
   }
 
+  async _toggleSkipFileStatus(params: Dap.ToggleSkipFileStatusParams): Promise<Dap.ToggleSkipFileStatusResult> {
+    if (params.resource) {
+      const absPath = urlUtils.fileUrlToAbsolutePath(params.resource);
+      if (absPath) {
+        params.resource = absPath;
+      }
+    }
+
+    if (!this.sourceContainer.scriptSkipper) {
+      if (!this._thread || !this._thread.cdp()) {
+        return {};
+      }
+      this.sourceContainer.initializeScriptSkipper(new ScriptSkipper([]));
+      this.sourceContainer.scriptSkipper!.setBlackboxSender(this._thread.cdp().Debugger);
+    }
+
+    const dapSource = await this._isSourceInCurrentStack(params);
+    if (dapSource && params.resource) {
+      const source = this.sourceContainer.source(dapSource);
+      if (source) {
+        source._blackboxed = !source._blackboxed;
```

Comment:
Not clear to me why you toggle a property on the Source, then call scriptSkipper.toggle which toggles another property in a map. Ideally the state would just be kept in one place

Reply:
I think you could get rid of the `_blackboxed` property and have the ScriptSkipper be the source of truth, and have the Source able to query the ScriptSkipper as needed?